### PR TITLE
Remove unused task action SegmentLockReleaseAction

### PR DESCRIPTION
**Pull request by kfaraz**
Changes:
- Remove `SegmentLockReleaseAction` as it is not used anywhere. It is not even registered as a known sub-type of `TaskAction`.
- Minor refactor in `TaskLockbox`. No functional change.
- Remove `ExpectedException` from `TaskLockboxTest`

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java [hunk 1]
```diff
@@ -1271,33 +1215,63 @@ public void remove(final Task task)
...
     }
-    finally {
-      giant.unlock();
+    catch (Exception e) {
+      log.warn(e, "Failure cleaning up upgradeSegments or pendingSegments tables.");
     }
   }
 
+  /**
+   * Finds the currently-active lock posses for the task.
```
Comment (AmatyaAvadhanula) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java:1256]: "currently-active" can be ambiguous as the method returns possses for both active and revoked locks

**Review by AmatyaAvadhanula — APPROVED** [review]
Left a minor comment, but otherwise LGTM.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java (hunk 1 above)
Comment (kfaraz) [reply indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java]: Fixed.

**Comment by kfaraz** [comment]
Thanks for the review, @AmatyaAvadhanula !

**Pull request merged**